### ISSUE TITLE
Spline 0.4 documentation

### ISSUE THREAD

**Issue by wajda** [AbsaOSS/spline#437]
1. Check all the READMEs and restructure if needed
1. Create GH pages for v0.4 (v0.3 should be retained)

**Comment by wajda** [comment]
Add NodeJS into the build requirements.

_See @kaustubhghode's post https://github.com/AbsaOSS/spline/issues/454#issuecomment-552912838_

**Comment by benedeki** [comment]
Some late comments on the PR:
* Line 14:  Two small changes (in both blue boxes), that I would consider making it clearer:
  * Spark -> Spark Application - the Spark Agent is not part of Spark itself IMO
  * Arango -> Arango DB - the Arango doesn't have to be known to all audiences. Adding the DB will clarify on the spot, it's a database, without the need to search for it.
* Line 22: I would reword it here in a way, that it's clear, starting which version the lineage is full covered.
* Line 70: Question: Why zip and not git? Or why to be particular at all? Could be just: Get the source from https://github.com/AbsaOSS/spline/
* **Start Spline Server** and **Start Spline UI** chapters should have siilar structure:
  * Docker
  * WAR deployment
  * Extra (for example t's not clear if the NodeJS part belongs to WAR deployment or is standalone)
* Line 151: Chapter **Check the result in the browser** I beleive is too high in the heararchy, it should be a sub-chapter of **Start Spline UI**
* Line 186: Tiny: Typography of this seems off. Using *`DISABLED`* feels more pleasing to eyes.
